### PR TITLE
MULE-8906: Implement simple connection support in Ext-Api

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: src/main/java/org/mule/extension/annotations/connector/Connector.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.mule.extension.annotations.connector;
```
Comment (pablokraan) [inline src/main/java/org/mule/extension/annotations/connector/Connector.java]: Add license header.

Comment (pablokraan) [inline src/main/java/org/mule/extension/annotations/connector/Connector.java]: Do we have the license check plugin on the extensions-api projects?

File: src/main/java/org/mule/extension/annotations/param/Connection.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package org.mule.extension.annotations.param;
```
Comment (pablokraan) [inline src/main/java/org/mule/extension/annotations/param/Connection.java]: Add license header

File: src/main/java/org/mule/extension/annotations/param/Connection.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.lang.reflect.Method;
+
+/**
+ * Signals that an argument of a {@link Method} annotated with {@link Operation} should be assigned
```
Comment (pablokraan) [inline src/main/java/org/mule/extension/annotations/param/Connection.java]: Signals > Indicates

File: src/main/java/org/mule/extension/annotations/connector/Connector.java (hunk 1 above)
Comment (marianogonzalez) [inline src/main/java/org/mule/extension/annotations/connector/Connector.java]: yes but for some reason it seems to not be validating at compile time. I'll tackle that in a separate PR. for now, just adding the license headers

File: src/main/java/org/mule/extension/annotations/param/Connection.java (hunk 3 above)
Comment (marianogonzalez) [inline src/main/java/org/mule/extension/annotations/param/Connection.java]: fixed

**Pull request merged**